### PR TITLE
refactoring: add redux to counter

### PR DESCRIPTION
**Pull request by sonokougg**
This change is [Reviewable](https://reviewable.io/reviews/tigerscave/sono-redux-counter/1)

### CODE REVIEW

**Comment by sonokougg** [comment]
@cdra0624 @zigen

File: src/reducers/reducer.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      }
+    }
+    case 'COUNT_DOWN': {
+      const { count } = state;
+      return {
+        count: count - 1
+      }
+    }
+    case 'COUNT_RESET': {
+      const { count } = state;
```
Comment (zigen) [inline src/reducers/reducer.js:25]: this `count` is not needed.

File: src/reducers/reducer.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import { COUNT_UP } from '../actionTypes.js';
+import { COUNT_DOWN } from '../actionTypes.js';
+import { COUNT_RESET } from '../actionTypes.js';
+
+const initialState = {
+  count: 0,
+}
+
+export default function (state = initialState, action) {
+  console.log(state);
```
Comment (zigen) [inline src/reducers/reducer.js:10]: remove `console.log` , since you use redux devtool.

File: src/reducers/reducer.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  switch (action.type) {
+    case 'COUNT_UP': {
```
Comment (zigen) [inline src/reducers/reducer.js:12]: use `actionType.COUNT_UP`, don't use string literal

File: src/reducers/index.js [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import { combineReducers } from 'redux';
+import reducer from './reducer';
+
+const rootReducer = combineReducers({
+  count: reducer,
```
Comment (zigen) [inline src/reducers/index.js:5]: the name `reducer` is not good. `count` or ` counter` is better.
you should change the file name `reducer.js` to `counter.js`

File: src/components/App.js [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+          <button onClick={this.props.countDown}>-</button>
+          <button onClick={this.props.countReset}>reset</button>
+        </div>
+      </div>
+    );
+  }
+}
+
+const mapStateToProps = (state, ownProps) => {
+  console.log(state);
```
Comment (zigen) [inline src/components/App.js:48]: remove `console.log`

File: src/components/App.js [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+
+    this.countReset = () => {
+      this.setState({
+        count: 0
+      })
+    }
+  }*/
+
+  render() {
+    const { count } = this.props;
```
Comment (zigen) [inline src/components/App.js:33]: also assign `countUp`, `countDown`, `countReset` here.

File: src/components/App.js [hunk 7]
```diff
@@ -0,0 +1,64 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { countUp, countDown, countReset } from '../actions/countAction';
+
+class App extends Component {
+/*  constructor(props) {
```
Comment (zigen) [inline src/components/App.js:6]: do not comment out previous code.
why do you use git????

File: src/actions/countAction.js [hunk 8]
```diff
@@ -0,0 +1,27 @@
+import { COUNT_UP } from '../actionTypes';
+import { COUNT_DOWN } from '../actionTypes';
+import { COUNT_RESET } from '../actionTypes';
+
+export function countUp() {
+  return dispatch => {
```
Comment (zigen) [inline src/actions/countAction.js:6]: no need to return function which takes `dispatch`.
this function should return action object

**Review by zigen — CHANGES_REQUESTED** [review]

**Comment by sonokougg** [comment]
I close this PR

**Comment by sonokougg** [comment]
This feature moves to #2

**Pull request closed without merging**